src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,213 @@
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> newNode;
+        if ((index > size) || (index < 0)) {
+            throw new IndexOutOfBoundsException("Can't add element - index bigger than size");
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index == 0) {
+            newNode = new Node<>(null, value, head);
+            head.prev = newNode;
+            head = newNode;
+            size++;
+            return;
+        }
+        if (index == size - 1) {
+            addNode(tail.prev, value);
+            size++;
+            return;
+        }
+        Node<T> oldNode;
+        oldNode = getNode(index);
+        addNode(oldNode, value);
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T listElement: list) {
+            add(listElement);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if ((index >= size) || (index < 0)) {
+            throw new IndexOutOfBoundsException("Can't get element - index bigger than size");
+        }
+        if (index == 0) {
+            return head.element;
+        }
+        if (index == size - 1) {
+            return tail.element;
+        }
+        return getNode(index).next.element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        T element;
+        if ((index >= size) || (index < 0)) {
+            throw new IndexOutOfBoundsException("Can't set element - index bigger than size");
+        }
+        if (index == 0) {
+            element = head.element;
+            head.element = value;
+            return element;
+        }
+        if (index == size - 1) {
+            element = tail.element;
+            tail.element = value;
+            return element;
+        }
+        Node<T> node;
+        node = getNode(index).next;
+        element = node.element;
+        node.element = value;
+        return element;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T element;
+        if ((index >= size) || (index < 0)) {
+            throw new IndexOutOfBoundsException("Can't remove element - index bigger than size");
+        }
+        if ((index == 0) && (size == 1)) {
+            head = null;
+            tail = null;
+            size--;
+            return null;
+        }
+        if (index == 0) {
+            element = head.element;
+            head = head.next;
+            head.prev = null;
+            size--;
+            return element;
+        }
+        if (index == size - 1) {
+            element = tail.element;
+            tail = tail.prev;
+            tail.next = null;
+            size--;
+            return element;
+        }
+        Node<T> oldNode;
+        oldNode = getNode(index).next;
+        element = oldNode.element;
+        unLink(oldNode);
+        size--;
+        return element;
```

Comment:
lets just find our node through `getNode(int index)` method and unlink it with `unLink(Node<T> oldNode)` method